emission/analysis/modelling/tour_model/tour_model_matrix.py

```diff
@@ -0,0 +1,275 @@
+# Our imports
+import emission.simulation.markov_model_counter as esmmc
+from emission.core.our_geocoder import Geocoder
+
+# Standard imports
+import numpy as np
+import math, datetime, heapq
+import networkx as nx
+import matplotlib.pyplot as plt
+
+# Constants
+DAYS_IN_WEEK = 7
+HOURS_IN_DAY = 24
+MODES_TO_NUMBERS = {"walking" : 0, "car" : 1, "train" : 2, "bart" : 3, "bike" : 4}
+NUM_MODES = len(MODES_TO_NUMBERS)
+
+class Commute(object):
+    
+    """ An edge in the graph """
+
+    def __init__(self, starting_point, ending_point):
+        self.probabilities = np.zeros((DAYS_IN_WEEK, HOURS_IN_DAY))
+        self.starting_point = starting_point
+        self.ending_point = ending_point
+        self.trips = [ ]
+
+    def increment_prob(self, hour, day):
+        self.probabilities[day, hour] += 1
+
+    def set_predicted_mode(self, pred_mode):
```

Comment:
Why do we need the predicted mode and confirmed mode for this? Aren't these the fake trips? How can you confirm the mode of a fake trip?